File: app/src/main/java/com/thanksmister/bitcoin/localtrader/ui/adapters/MethodAdapter.java
/*
 * Copyright (c) 2018 ThanksMister LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed
 * under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package com.thanksmister.bitcoin.localtrader.ui.adapters;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.thanksmister.bitcoin.localtrader.R;
import com.thanksmister.bitcoin.localtrader.persistence.Method;

import java.util.Collections;
import java.util.List;

public class MethodAdapter extends ArrayAdapter<Method> {
    private Context context;
    private List<Method> items = Collections.emptyList();

    public MethodAdapter(Context _context, int _resource, List<Method> _items) {
        super(_context, _resource, _items);
        context = _context;
        items = _items;
    }

    @Override
    public int getCount() {
        if (items == null) return 0;
        return items.size();
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            LayoutInflater mInflater = (LayoutInflater) context.getSystemService(Activity.LAYOUT_INFLATER_SERVICE);
            convertView = mInflater.inflate(R.layout.spinner_transactions_header_layout, null);
        }

        if (position < getCount()) {
            TextView spinnerTarget = (TextView) convertView.findViewById(R.id.spinnerTarget);
            spinnerTarget.setText(items.get(position).getName());
        }

        return convertView;
    }

    @Override
    public View getDropDownView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            LayoutInflater mInflater = (LayoutInflater) context.getSystemService(Activity.LAYOUT_INFLATER_SERVICE);
            convertView = mInflater.inflate(R.layout.spinner_layout, null);
        }

        if (position < getCount()) {
            TextView spinnerTarget = (TextView) convertView.findViewById(R.id.spinnerTarget);
            spinnerTarget.setText(items.get(position).getName());
        }

        return convertView;
    }
}

File: app/src/main/java/com/thanksmister/bitcoin/localtrader/ui/adapters/MethodAdapter.kt
/*
 * Copyright (c) 2020 ThanksMister LLC
 *  http://www.thanksmister.com
 *
 *  Mozilla Public License 2.0
 *
 *  Permissions of this weak copyleft license are conditioned on making
 *  available source code of licensed files and modifications of those files
 *  under the same license (or in certain cases, one of the GNU licenses).
 *  Copyright and license notices must be preserved. Contributors provide
 *  an express grant of patent rights. However, a larger work using the
 *  licensed work may be distributed under different terms and without source
 *  code for files added in the larger work.
 */

package com.thanksmister.bitcoin.localtrader.ui.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

import com.thanksmister.bitcoin.localtrader.R
import com.thanksmister.bitcoin.localtrader.network.api.model.Method

class MethodAdapter(context: Context, _resource: Int, _items: List<Method>) : ArrayAdapter<Method>(context, _resource, _items) {
    private var items = emptyList<Method>()
    private val inflater: LayoutInflater = LayoutInflater.from(context)

    init {
        items = _items
    }

    override fun getCount(): Int {
        return items.size
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val view = if (convertView != null) {
            convertView
        } else {
            inflater.inflate(R.layout.spinner_transactions_header_layout, parent, false)
        }
        if (position < count) {
            val spinnerTarget = view.findViewById<View>(R.id.spinnerTarget) as TextView
            spinnerTarget.text = items[position].name
        }
        return view
    }

    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View {
        val view = convertView ?: inflater.inflate(R.layout.spinner_layout, parent, false)
        if (position < count) {
            val spinnerTarget = view.findViewById<View>(R.id.spinnerTarget) as TextView
            spinnerTarget.text = items[position].name
        }
        return view
    }
}